src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordPrintMembers.cs

```diff
@@ -0,0 +1,277 @@
+                        return;
+                    }
+                    else
+                    {
+                        // if (base.print(builder))
+                        //     builder.Append(", ")
+                        block!.Add(F.If(basePrintCall, makeAppendString(F, builder, ", ")));
+                    }
+                }
+
+                Debug.Assert(!printableMembers.IsEmpty && block is object);
+
+                for (var i = 0; i < printableMembers.Length; i++)
+                {
+                    // builder.Append(<name>);
+                    // builder.Append(" = ");
+                    // builder.Append((object)<value>);
+                    // builder.Append(", "); // except for last member
+
+                    var member = printableMembers[i];
+                    block.Add(makeAppendString(F, builder, member.Name));
+                    block.Add(makeAppendString(F, builder, " = "));
+
+                    var value = member.Kind switch
+                    {
+                        SymbolKind.Field => F.Field(F.This(), (FieldSymbol)member),
+                        SymbolKind.Property => F.Property(F.This(), (PropertySymbol)member),
+                        _ => throw ExceptionUtilities.UnexpectedValue(member.Kind)
+                    };
+
+                    block.Add(F.ExpressionStatement(
+                        F.Call(receiver: builder,
+                            F.WellKnownMethod(WellKnownMember.System_Text_StringBuilder__AppendObject),
+                            F.Convert(F.SpecialType(SpecialType.System_Object), value))));
+
+                    if (i < printableMembers.Length - 1)
+                    {
+                        block.Add(makeAppendString(F, builder, ", "));
+                    }
+                }
+
+                block.Add(F.Return(F.Literal(true)));
+
+                F.CloseMethod(F.Block(block.ToImmutableAndFree()));
+            }
+            catch (SyntheticBoundNodeFactory.MissingPredefinedMember ex)
+            {
+                diagnostics.Add(ex.Diagnostic);
+                F.CloseMethod(F.ThrowNull());
+            }
+
+            static BoundStatement makeAppendString(SyntheticBoundNodeFactory F, BoundParameter builder, string value)
+            {
+                return F.ExpressionStatement(F.Call(receiver: builder, F.WellKnownMethod(WellKnownMember.System_Text_StringBuilder__AppendString), F.StringLiteral(value)));
+            }
+        }
+
+        internal static MethodSymbol? FindValidPrintMembersMethod(TypeSymbol containingType, CSharpCompilation compilation)
+        {
+            if (containingType.IsErrorType() || containingType.IsObjectType())
+            {
+                return null;
+            }
+
+            MethodSymbol? candidate = null;
+            var accessibility = (containingType.BaseTypeNoUseSiteDiagnostics.IsObjectType() && containingType.IsSealed)
+                ? Accessibility.Private
```

Comment:
>Accessibility.Private [](start = 18, length = 21)

Do we need to handle this case? If this method is used to find the definition of "PrintMembers" in the base type, if the base type is `sealed`, an error will be reported for deriving from a `sealed` type so we shouldn't need to find "PrintMembers" in that case. And we won't find "PrintMembers" if the 'sealed' type is from metadata anyway.

Reply:
That's a good simplification. Thanks

---
In reply to: [470159574](https://github.com/dotnet/roslyn/pull/46616#discussion_r470159574) [](ancestors = 470159574)